inc/templates.php

```diff
@@ -483,7 +575,15 @@ function hydrate_template_parts( $component ) {
 		return false;
 	}
 
-	return hydrate_components( prepare_data_from_template( $template ) );
+	$template_data = prepare_data_from_template( $template );
+
+	// Handle template parts with only one component rather than an array of
+	// components.
+	if ( isset( $template_data['name'] ) ) {
```

Comment:
`prepare_data_from_template()` should always return an array. If you're seeing a case where that doesn't happen, let's handle it within that function.

Reply:
It's returning an array, depending if the template part is an array (multiple components) vs. object (single component), then we need to wrap the single array in another array so it hydrates properly, otherwise it includes the template part weird.

I'll try to add a test for this.